plugins/lookup/merge_variables.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Thales Netherlands
+# Copyright (c) 2021, Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    author:
+      - Roy Lenferink (@rlenferink)
+      - Mark Ettema (@m-a-r-k-e)
+    name: merge_variables
+    short_description: merge variables with a certain suffix
+    description:
+        - This lookup returns the merged result of all variables in scope that matches the given suffix optionally
+          starting with an initial value.
+    version_added: 6.1.0
+    options:
+      _terms:
+        description:
+          - The suffix of the variable name (all available variables will be checked with endswith(suffix))
+        required: true
+      initial_value:
+        description:
+          - An initial value to start with
+        required: false
+      override_warning:
+        description:
+          - Print a warning when a key will be overwritten
+        default: false
+      override_error:
+        description:
+          - Return error when a key will be overwritten
+        default: false
+"""
+
+EXAMPLES = """
+# Some example variables, they can be defined anywhere as long as they are in scope
+test_init_list:
+  - "list init item 1"
+  - "list init item 2"
+
+testa__test_list:
+  - "test a item 1"
+
+testb__test_list:
+  - "test b item 1"
+
+testa__test_dict:
+  ports:
+    - 1
+
+testb__test_dict:
+  ports:
+    - 3
+
+
+# Merging variables that ends with '__test_dict' and store the result in a variable 'example_a'
+example_a: "{{ lookup('merge_variables', '__test_dict') }}"
```

Comment:
This example will not work, since you do not use the FQCN of the plugin.
```suggestion
example_a: "{{ lookup('community.general.merge_variables', '__test_dict') }}"
```
Same below.

Reply:
Done